### ISSUE TITLE
Add "Not directly translated into language" drop-down to "translate the sentences of this user"

### ISSUE THREAD

**Issue by alanfgh** [Tatoeba/tatoeba2#373]
Posted on the Wall by Silja: 

Suggestion: Please add "Not directly translated into language" drop-down to "translate the sentences of this user" (I mean this page: http://tatoeba.org/fin/activities/translate_sentences_of/Silja)

**Comment by RyckRichards** [comment]
Requested once more by CK during the 1st Tatoeba day - 

https://tatoeba.org/eng/wall/show_message/22943#message_22943 
Answer (by Trang): https://tatoeba.org/eng/wall/show_message/22982#message_22982

**Comment by ckjpn** [comment]
Perhaps this could somehow be interfaced with the advanced search we now have.

All you would need to do is have the following URLs created.

English Sentences by CK not translated into any language.

https://tatoeba.org/eng/sentences/search?user=CK&trans_filter=exclude&sort=random&from=eng

English Sentences by CK not directly translated into French

https://tatoeba.org/eng/sentences/search?user=CK&trans_filter=exclude&sort=random&from=eng&to=fra



You could possibly just close this issue because this is something that is now possible for members to do through the advanced search.  However, it's likely that many members, especially new members, don't know this is possible, so it might encourage more translations of sentences by native speakers which would likely improve the quality of the corpus.

**Comment by ckjpn** [comment]
> alanfgh added the priority:low ....

I would suggest this should be a high priority.
It should be fairly easy to do, too.  See the above comment on using the "advanced search" to do this.

**Comment by agrodet** [comment]
I think this is partly duplicated by #683 where @alanfgh asked not only the "Not directly translated into" but also the "With audio" setting.

In #683, #627 is mentioned. I was thinking of checking all this, including the possibility to add the "With audio" on the "Translate sentences" page. However, I saw that @trang added #627 to the Responsive UI project. Am I correct to assume that we should wait for trang to implement its solution before dealing with the other ones (this one included)?

**Comment by trang** [comment]
@agrodet You don't necessarily need to wait for #627 to be solved. 

I added #627 in the Responsive UI project because it was a good occasion to figure out a design pattern for sidebar filters/options in the responsive UI. 

On a mobile screen, we're not going to have this "main section" and "annexe section" anymore. We will only have a single column. The filters will have to be in a sidenav that opens when clicking on some icon.
On a desktop screen, this sidenav should be displayed by default, unless the user decides to shrink the width of their browser.

I wanted to use #627 as a starting point to introduce this behavior, that's why it's been added into the Responsive UI backlog.

For this ticket here, you could ignore the responsive aspect and simply work on adding one more dropdown and have the proper database query behind it. But maybe adding this dropdown is not the best solution.

As CK mentioned above:

> this is something that is now possible for members to do through the advanced search

So we have other possibilities:
- We can change the URL of "Translate sentences of XXX" to redirect to the search with `owner=XXX`.
- We can add an additional field in the "Translate sentences" page so that contributors can specify the owner.

**Comment by agrodet** [comment]
For all roads leading to Rome, we need roads :) 

We could change many pages to redirect to the results page of an advanced search. I think that's not a good experience at all. Even if the way to display three different pages is the same, if the purpose of these three pages is different, then we should have three pages.  
Let's take the example of "Translate AlanF_US sentences". My goal is to translate Alan's sentences. I'm happy with a page named "Translate Alan's sentences", less happy with a page named "Search results (1000 among 12000)".  
Similarly, the "Translate sentences" has its role to play as a "goal-oriented" middlepage. We could remove that page arguing that the same can be done with an advanced search. But the purpose behind "searching" is different from the purpose behind "translating".

Obviously, one could argue that users used to the website take shortcuts, and don't use these pages anymore. But that is up to them. A design that allow shortcuts is better than one that forces you to use it the only way it decided.

Of course, all that is only my personal view of design and can (should) be debated. (Redirections are evil, ssshhh)

---

Trang, thank you for explaining the situation, I understand now. Maybe I will open a PR in the next few days with some of the changes mentioned in all these issues I think relevant and why.

**Comment by trang** [comment]
> We could change many pages to redirect to the results page of an advanced search. I think that's not a good experience at all.

I agree, it would definitely not be our target but it would remain consistent with what we provide at the moment (for the fact that our "Translate sentences" page redirects to the search). 

I personally don't like that the "Translate sentences" page redirects to the search. I believe that it should be reworked into some sort of translation workspace. I briefly mentioned this before (cf. https://github.com/Tatoeba/tatoeba2/issues/1351#issuecomment-436521014 and #1700). 

Right now the search is the temporary solution for the translation workspace because it's the only place where a user can define precisely what they want to translate. But this translation workspace should be a section on its own, that is subject to different requirements than the search. It would be optimized for translation activity.

You can go directly to this translation workspace if you're in the mood for translating. Otherwise, everywhere else where you could be tempted to start translating, you would have a shortcut to the translation workspace. It would prefill the translation workspace with some criteria, based on where you came from. You can refine these criteria as you translate.

The page "Translate sentences of this user" is not the only page where people asked to be able to filter by sentences not directly translated into a certain language. This was requested for the lists (#611), for the "several random sentences" page (#237), and I can imagine people might want to request it for the tags or the audio as well.
